src/main/java/com/auth0/exception/APIException.java

```diff
@@ -81,6 +81,49 @@ public String getDescription() {
         return description;
     }
 
+    /**
+     * @return {@code true} when an MFA code is required to authenticate, {@code false} otherwise.
+     */
+    public boolean isMultifactorRequired() {
+        return "mfa_required".equals(error);
+    }
+
+    /**
+     * @return {@code true} when the username and/or password used for authentication are invalid, {@code false} otherwise.
+     */
+    public boolean isInvalidCredentials() {
+        return "invalid_user_password".equals(error) || "invalid_grant".equals(error) && "Wrong email or password.".equals(description);
+    }
+
+    /**
+     * @return {@code true} when MFA is required and the user is not enrolled, {@code false} otherwise.
+     */
+    public boolean isMultifactorEnrollRequired() {
+        return "unsupported_challenge_type".equals(error);
+    }
+
+    /**
+     * @return {@code true} when Bot Protection flags the request as suspicious, {@code false} otherwise.
+     */
+    public boolean isVerificationRequired() {
+        return "requires_verification".equals(error);
+    }
+
+    /**
+     * @return {@code true} when the MFA Token used on the login request is malformed or has expired, {@code false} otherwise.
+     */
+    public boolean isMultifactorTokenInvalid() {
+        return "expired_token".equals(error) && "mfa_token is expired".equals(description) ||
+                "invalid_grant".equals(error) && "Malformed mfa_token".equals(description);
+    }
+
+    /**
+     * @return {@code true} when authenticating with web-based authentiction and the resource server denied access per the OAuth 2 spec, {@code false} otherwise.
+     */
+    public boolean isAccessDenied() {
```

Comment:
I remember the code for "user is blocked" (too many attempts, or manually blocked) but this one I can't remember the use case. Do you? Otherwise, it maybe best to leave out of this PR.

Reply:
No. I borrowed it from from [here](https://github.com/auth0/Auth0.Android/blob/master/auth0/src/main/java/com/auth0/android/authentication/AuthenticationException.java#L232) in Auth0.Android, and it is listed as a standard error on the [docs](https://auth0.com/docs/api/authentication#standard-error-responses).